modules/core/src/test/java/org/apache/ignite/internal/processors/service/IgniteServiceDeploymentClassLoadingDefaultMarshallerTest.java

```diff
@@ -152,6 +163,30 @@ public void testServiceDeployment3() throws Exception {
         startClientGrid(CLIENT_NODE_WITH_EXT_CLASS_LOADER).services().deploy(serviceConfig());
     }
 
+    /** @throws Exception If failed. */
+    @Test
+    public void testFailWhenNodeFilterClassNotFound() throws Exception {
```

Comment:
Suggestion: a test for re-deploy. Or several. Scenarios: redeploy with empty node filter, redeploy with a known class filter, redeploy with another unknows class ode filter. WDYT?